### PR TITLE
Missing hooks + Hook aliases as title (better search results in Algolia)

### PR DESCRIPTION
**Pull request by thomasnares**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 8.x
| Description?  | Large refresh of hook referential ,with a shortcode to render the page (and allow search/fiiltering via the YAML content)

**NEED THIS PR :** https://github.com/PrestaShop/ps-docs-theme/pull/46

With this kind of markdown: 

```
---
menuTitle: displayAdminGridTableAfter
Title: displayAdminGridTableAfter
hidden: true
hookTitle: 'Display after Grid table'
files:
    -
        url: 'https://github.com/PrestaShop/PrestaShop/blob/8.0.x/src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig'
        file: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig
locations:
    - 'back office'
type: display
hookAliases:
    - displayAdminListAfter
array_return: false
check_exceptions: false
chain: false
origin: core
description: 'This hook adds new blocks after Grid component table'

---

{{% hookDescriptor %}}
```

You obtain (with the shortcode) :

Screenshot 2023-10-06 at 18 17 30

### CODE REVIEW

**Comment by thomasnares** [comment]
Needs https://github.com/PrestaShop/ps-docs-theme/pull/46 to work

**Comment by thomasnares** [comment]
@kpodemski ready for review, good luck 😅

**Comment by thomasnares** [comment]
And added the documentation for the documentation 🧠 

https://github.com/PrestaShop/docs/pull/1733/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5

File: README.md [hunk 1]
```diff
@@ -23,3 +23,57 @@ The deployment pipeline is documented on the [sources repository for DevDocs](ht
...
 
 Content from this documentation is licensed under the [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/).
+
+## Maintaining Hook list referential
+
+A complete referential of hook and informations on each hook is located in `modules/concepts/hooks/list-of-hooks/`. 
+
+This referential can be generated with a script, crawling all: 
+
+- Prestashop (core) codebase
```
Comment (kpodemski) [inline README.md]:
```suggestion
- PrestaShop (core) codebase
```

File: README.md [hunk 2]
```diff
@@ -23,3 +23,57 @@ The deployment pipeline is documented on the [sources repository for DevDocs](ht
...
+- all native modules
+- all native themes
+
+In the documentation, a hook is described with in yaml format, with this format:
```
Comment (kpodemski) [inline README.md]:
```suggestion
In the documentation, a hook is described in yaml format: 
```

File: README.md [hunk 3]
```diff
@@ -23,3 +23,57 @@ The deployment pipeline is documented on the [sources repository for DevDocs](ht
 ## License
 
 Content from this documentation is licensed under the [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/).
+
+## Maintaining Hook list referential
+
+A complete referential of hook and informations on each hook is located in `modules/concepts/hooks/list-of-hooks/`. 
+
+This referential can be generated with a script, crawling all:
```
Comment (kpodemski) [inline README.md:31]: A script that we can put somewhere to the repo?

File: README.md [hunk 4]
```diff
@@ -23,3 +23,57 @@ The deployment pipeline is documented on the [sources repository for DevDocs](ht
...
+type: string #action or display
+hookAliases:
+    - string #old hook name or alias
+array_return: boolean #true or false if the hook has an `$array_return` parameter set to `true`
+check_exceptions: boolean #true or false if the hook has a `$check_exceptions` parameter set to `false`
+chain: boolean #true or false if the hook has a `$chain` parameter set to `true` 
+---
+```
+
+This `yaml` structure allows the shortcode `hookDescriptor` to render a digest html page for the documentation, and allows the search script to search in this metadata.
```
Comment (kpodemski) [inline README.md]:
```suggestion
This YAML structure enables the shortcode `hookDescriptor` to generate an HTML page with condensed information for documentation, and allows the search script to search within this metadata.
```

File: README.md [hunk 5]
```diff
@@ -23,3 +23,57 @@ The deployment pipeline is documented on the [sources repository for DevDocs](ht
...
+
+This `yaml` structure allows the shortcode `hookDescriptor` to render a digest html page for the documentation, and allows the search script to search in this metadata. 
+
+Next, add the shortcode in the page's content: 
+
+```
+{{% hookDescriptor %}}
+```
+
+Finally, if available, add a section with a code example, and parameters details:
```
Comment (kpodemski) [inline README.md]:
```suggestion
Finally, if available, add a section with a code example and parameters details: 
```

File: modules/concepts/hooks/list-of-hooks/action<DefinitionId>GridDataModifier.md [hunk 6]
```diff
@@ -1,34 +1,30 @@
 ---
-menuTitle: action<DefinitionId>GridDataModifier
 Title: action<DefinitionId>GridDataModifier
 hidden: true
 hookTitle: 
 files:
-  - src/Core/Grid/GridFactory.php
+    -
+        url: 'https://github.com/PrestaShop/PrestaShop/blob/8.0.x/src/Core/Grid/GridFactory.php'
```
Comment (kpodemski) [inline modules/concepts/hooks/list-of-hooks/action<DefinitionId>GridDataModifier.md:7]: One idea. Could we do something like this?

```
url: https://github.com/PrestaShop/PrestaShop/blob/8.0.x/
path: src/Core/Grid/GridFactory.php
```

url could be updated from one single place, path is unique, wdyt? are there any corner cases where this cannot work?

Comment (thomasnares) [reply modules/concepts/hooks/list-of-hooks/action<DefinitionId>GridDataModifier.md:7]: You want to put `https://github.com/PrestaShop/PrestaShop/blob/8.0.x/` globally and not on each page ? 

- sometimes we may need to target a different branch (8.1.x, ...)
- sometimes the url is related to a module or a theme

So we need it on each page. 

We could reduce the url to `https://github.com/PrestaShop/PrestaShop/blob/8.0.x/`, and build the complete URL with the path, but is it worth it ?

File: README.md (hunk 3 above)
Comment (kpodemski) [reply README.md:31]: as we spoke, it could be added as a Gist

Comment (thomasnares) [reply README.md:31]: https://gist.github.com/thomasnares/b0605fdd9f5e77fa16bf5a42408feba3

I need to do some rewrite to fit the new format

Comment (kpodemski) [reply README.md:31]: let's add a task to link `with a script` to this file once it's rewritten to the new format, ok?

Comment (thomasnares) [reply README.md:31]: yes 👍

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
magic button clicked, let's see the results in a few minutes 🎉 thanks @thomasnares

**Pull request merged**